modules/src/core/ics05_port/context.rs

```diff
@@ -1,11 +1,68 @@
+use alloc::format;
+
 use crate::core::ics05_port::capabilities::Capability;
 use crate::core::ics05_port::error::Error;
 use crate::core::ics24_host::identifier::PortId;
+use crate::core::ics24_host::path::PortsPath;
+
+/// A context supplying all the necessary read-only dependencies for processing any information regarding a port.
+pub trait PortReader: CapabilityReader {
+    /// Module Id type that can be mapped to an ICS26 router callback module
+    type ModuleId;
+
+    /// Return the module_id along with the capability associated with a given port_id
+    fn lookup_module_by_port(
+        &self,
+        port_id: &PortId,
+    ) -> Result<(Self::ModuleId, Capability), Error>;
+
+    /// Check if the specified port_id is already bounded
+    fn is_bound(&self, port_id: PortId) -> bool {
+        self.get_capability(format!("{}", PortsPath(port_id)))
+            .is_ok()
+    }
+
+    /// Authenticate a capability key against a port_id by checking if the capability was previously
+    /// generated and bound to the specified port
+    fn authenticate(&self, port_id: PortId, capability: &Capability) -> bool {
+        self.authenticate_capability(format!("{}", PortsPath(port_id)), capability)
```

Comment:
Can we just do `to_string()` here instead?

```suggestion
        self.authenticate_capability(PortsPath(port_id).to_string(), capability)
```